### PR TITLE
Lab 11

### PR DESCRIPTION
**Pull request by TimM32**

### CODE REVIEW

File: driver/index.js [hunk 1]
```diff
@@ -1,23 +1,8 @@
...
-    
-    eventEmitter.emit('EVENT', 'delivered', payload);
-  }, 500);
-};
-  
-module.exports = { pickupHandler, intransitHandler };
\ No newline at end of file
+eventEmitter.on('Collect', collectPackage);
+eventEmitter.on('Moving', movingPackage);
+eventEmitter.on('Delivered', deliveredPackage);
```
Comment (DSFrey) [inline driver/index.js:8]: What is emitting these events?

File: driver/handler.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+const movingPackage = (payload) => {
+    setTimeout(() => {
+        eventEmitter.emit('Package received', payload);
+    }, 4000);
+};
+
+const deliveredPackage = (payload) => {
+    console.log('Driver: received', payload.orderId);
+    eventEmitter.emit('delivered', payload);
```
Comment (DSFrey) [inline driver/handler.js]: You're making an infinite loop here.

File: vendor/index.js [hunk 3]
```diff
@@ -1,19 +1,11 @@
...
-    setTimeout(() => {
-        console.log('VENDOR: Thank you for your order', payload.customer);
-        eventEmitter.emit('DELIVERED', payload);
-      }, 500);
-}
-
-module.exports = { newOrderHandler, deliveredHandler };
\ No newline at end of file
+eventEmitter.on('Package ready', orderHandler);
+eventEmitter.on('received', orderMessage);
```
Comment (DSFrey) [inline vendor/index.js]: Again, what is emitting this event?

File: vendor/handler.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const Chance = require('chance');
+const eventEmitter = require('../event-pool');
+
+let chance = new Chance();
+
+const orderHandler = (payload = null) =< {
```
Comment (DSFrey) [inline vendor/handler.js]: Arrow function syntax mistake

File: driver/handler.js (hunk 2 above)
Comment (TimM32) [reply driver/handler.js]: I think I fixed it with by changing line 5 in to: 
const collectPackage = (payload = null) => {

**Pull request merged**